Mathlib/Logic/Equiv/TransferInstance.lean

```diff
@@ -600,6 +600,35 @@ noncomputable def _root_.Shrink.linearEquiv [Small.{v} α] [AddCommMonoid α] [M
     Shrink.{v} α ≃ₗ[R] α :=
   Equiv.linearEquiv _ (equivShrink α).symm
 
+variable {R : Type*} [CommSemiring R]
+variable (A : Type*) [Semiring A] [Algebra R A]
+variable [AddCommMonoid α] [AddCommMonoid β] [Module A β]
+
+/-- Transport a module instance via an isomorphism of the underlying abelian groups.
+This has better definitional properties than `Equiv.module` since here
+the abelian group structure remains unmodified. -/
+def _root_.AddEquiv.module (e : α ≃+ β) :
```

Comment:
This needs to be an `abbrev`; it's used in other typeclasses, e.g. IsScalarTower, as a parameter.

Reply:
Changed, but could you please explain why this has to be reducible if it is used as a parameter in a typeclass term?

Reply:
See the library note [here](https://github.com/leanprover-community/mathlib4/blob/16956ef06e49e3e4f34d7be6d97f170580b42a79/Mathlib/Algebra/HierarchyDesign.lean#L190-L205); unfortunately notes don't seem to appear in the documentation currently.